gradio-app/gradio
# Share button

Creates shareable button attached to Image, Gallery, Chatbot, Audio, Video.
Automatically set to True on Spaces. See example code below.

To test locally:
1) replace `https://huggingface.co/uploads` with `https://radames-temp-proxy.hf.space/https://huggingface.co/uploads` in utils.ts (to avoid CORS error)
2) no dev mode, need to `pnpm run build` to avoid CORS error
3) replace kitchen_sink/run.py with code below (to use example files)

Should also test on spaces using built package for gradio version (can be found in requirements.txt of https://huggingface.co/spaces/gradio-pr-deploys/pr-4651-all-demos)


![Screen Shot 2023-06-26 at 11 22 49 PM](https://github.com/gradio-app/gradio/assets/7870876/312084b0-4943-4177-85e6-e7845ebea7fc)



```python
import gradio as gr
import os
os.environ['SYSTEM'] = "spaces"
os.environ['SPACE_ID'] = "aliabid94/audiotest100"


with gr.Blocks() as demo:
    with gr.Row():
        with gr.Column():
            text = gr.Textbox()
            num = gr.Slider()
            radio = gr.Radio(["a", "b", "c"], value="a")
            file = gr.File()
            btn = gr.Button("Run")
        with gr.Column():
            img = gr.Image()
            audio = gr.Audio()
            video = gr.Video()
            gallery = gr.Gallery()
            chatbot = gr.Chatbot()

    btn.click(
        lambda *args: [
            "files/lion.jpg",
            "files/cantina.wav",
            "files/world.mp4",
            ["files/lion.jpg", "files/tower.jpg"],
            [["Hey", "I'm bot"], ["I'm human", "Ok"]],
        ],
        [text, num, radio, file],
        [img, audio, video, gallery, chatbot],
    )

if __name__ == "__main__":
    demo.launch()
```  

Closes: #2749
